packages/solana-wallet-injection/src/wallet.ts

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable no-console */
+import { Liquality, LiqualityEvent } from '@liquality/solana-wallet-standard/src/window';
+import {
+  PublicKey,
+  Transaction,
+  VersionedTransaction,
+  SendOptions,
+  Connection,
+  Message,
+  Keypair,
+} from '@solana/web3.js';
+import { SolanaParser, ParsedInstruction } from '@debridge-finance/solana-transaction-parser';
+import { Idl } from '@project-serum/anchor';
+import { COMMON_REQUEST_MAP } from './types';
+import { BigNumber } from '@chainify/types';
+import base58 from 'bs58';
+
+/**
+ * @dev users are paying a fixed fee of 5000 lamports per transaction/signature
+ */
+const DEFAULT_FEE = '0.000005';
+
+export class LiqualitySolanaWallet implements Liquality {
+  public publicKey: PublicKey | null;
+  public isConnected: boolean;
+  private window: Window;
+  private _connection: Connection;
+
+  public get connection(): Connection {
+    return this._connection;
+  }
+  public set connection(value: Connection) {
+    this._connection = value;
+  }
+  private solanaParser = new SolanaParser([]);
+
+  constructor(window: Window) {
+    this.window = window;
+    // TODO: take rpc url from cryptoassets
+    this.connection = new Connection(
+      'https://red-sleek-rain.solana-mainnet.discover.quiknode.pro/fc112deb1e0228d09d0d8c12b8de5a601d251d80'
+    );
+  }
+
+  public async connect(
+    _options?: { onlyIfTrusted?: boolean | undefined } | undefined
+  ): Promise<{ publicKey: PublicKey }> {
+    const { accepted } = await this.enable();
+
+    if (!accepted) {
+      throw new Error('User rejected');
```

Comment:
@kraikov Please use the error from error parser. [See example](https://github.com/liquality/wallet/blob/cd1982dc5bc9d38f0ed1e7e8e89f9667d4f58755/src/pageProvider/solanaProvider.js#:~:text=if%20(!accepted)%20throw%20new%20UserDeclinedError())